### PR TITLE
issues/101 Fixed state setting asynchronously

### LINKED ISSUE #101: State setting not happening asynchronously
There is an issue with the getState function on XAPI that is resulting in the xapiwrapper.getState calls happening after the Model state has been set (to empty)

### PR DESCRIPTION
**Pull request by cahirodoherty-learningpool**
fixes #101 

Swapped out Async.each for a for/in loop to resolve async/await issues

### CODE REVIEW

**Review by eleanor-heath — APPROVED** [review]

**Review by ethan-lp — APPROVED** [review]

**Comment by oliverfoster** [comment]
Did it fix the problem? I can't see how the code is different from before, with the exception of https://github.com/adaptlearning/adapt-contrib-xapi/pull/102/files#diff-b0b5ab3ca1ad9e3f2e6a5f7f072256b8d933fdcdb7f6b929fd463257fde05229R1130 `if (!_.isEmpty(state)) {`

**Comment by cahirodoherty-learningpool** [comment]
> 

Hi Oli, yeah this fixed the issue. The await is now being respected and the state is getting populated correctly. I'll remove the async library here now

**Comment by cahirodoherty-learningpool** [comment]
@oliverfoster The Async library has now been removed

File: js/XAPI.js [hunk 1]
```diff
@@ -1404,12 +1397,14 @@ class XAPI extends Backbone.Model {
   async processAttachments(statement) {
     const attachments = statement.attachments;
 
-    await Async.each(attachments, (attachment, nextAttachment) => {
-
-      // First check the attachment for a value
+    for (let attachment of attachments) {
+      await new Promise((resolve, reject) => {
```
Comment (oliverfoster) [inline js/XAPI.js:1401]: Bad indentation inside this promise code block.

File: js/XAPI.js [hunk 2]
```diff
@@ -1404,12 +1397,14 @@ class XAPI extends Backbone.Model {
...
+        // First check the attachment for a value
       if (attachment.value) {
-        nextAttachment();
-      } else if (attachment.url) {
+        resolve();
```
Comment (oliverfoster) [inline js/XAPI.js:1404]: you'll have to return here otherwise it'll carry on into the attachment.url block

File: js/XAPI.js (hunk 1 above)
Comment (oliverfoster) [inline js/XAPI.js:1401]: bad indentation inside the promise code block

**Review by oliverfoster — CHANGES_REQUESTED** [review]

**Comment by cahirodoherty-learningpool** [comment]
@oliverfoster Indentation and return of resolve now fixed

**Review by oliverfoster — APPROVED** [review]

**Pull request merged**